main.go

```diff
@@ -118,22 +147,18 @@ func main() {
 
 	// Set default manager options
 	options := manager.Options{
-		Namespace:              namespace,
+		Metrics:                metricsServerOptions,
+		WebhookServer:          webhookServer,
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       "b674928d.argoproj.io",
 	}
 
-	// Add support for MultiNamespace set in WATCH_NAMESPACE (e.g ns1,ns2)
-	// Note that this is not intended to be used for excluding namespaces, this is better done via a Predicate
-	// Also note that you may face performance issues when using this with a high number of namespaces.
-	// More Info: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder
-	if strings.Contains(namespace, ",") {
-		options.Namespace = ""
-		options.NewCache = cache.MultiNamespacedCacheBuilder(strings.Split(namespace, ","))
+	if watchedNsCache := getDefaultWatchedNamespacesCacheOptions(); watchedNsCache != nil {
+		options.Cache = cache.Options{
+			DefaultNamespaces: watchedNsCache,
+		}
```

Comment:
I see below error when operator is installed via OLM. If I comment out above everything works fine. 
```
2023-10-25T05:49:04Z	INFO	controller_argocd	reconciling roles
2023-10-25T05:49:04Z	INFO	controller_argocd	unable to list: default because of unknown namespace for the cache
2023-10-25T05:49:04Z	ERROR	Reconciler error	{"controller": "argocd", "controllerGroup": "argoproj.io", "controllerKind": "ArgoCD", "ArgoCD": {"name":"argocd","namespace":"default"}, "namespace": "default", "name": "argocd", "reconcileID": "f074b983-3fcb-4c50-a37d-ec63f4c9e117", "error": "unable to list: default because of unknown namespace for the cache"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227
```

Reply:
fixed, thanks @svghadi